### DISCUSSION TITLE
Changing device Descriptor /manufacturer via Arduino IDE

### DISCUSSION THREAD

**Discussion by mayermakes** [hathach/tinyusb#1545] [General]
Hi, I see there is a possibility to change the device Descriptor /manufacturer for an ESP32S2 programmatically (instead of through idf.py menuconfig) I need to configure this in an Arduino sketch, but could not figure out how to do it.
Can someone point me in the right direction /tutorial for this?
thank you!

**Comment by pigrew** [comment]
I don't know the answer, and the arduino-esp32 guithub might be a better place to ask.

Looking at their code:

https://github.com/espressif/arduino-esp32/blob/8ee5f0a11e5423a018e0f89146e05074466274db/cores/esp32/USB.cpp#L122

It seems like their library sets the VID/PID when the USB object is created. TinyUSB does not support deinitialization, so it seems you cannot change it (with their existing implementation)  after the USB object is loaded. You might be able to set the `USB_VID` define to be a global variable of your choosing.

**Comment by hathach** [comment]
thanks @pigrew for answering the question. as always the code has the answer for all question. Look at the code if in doubt :)
